packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,259 @@
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `default`
+   */
+  size?: ModalSize;
+
+  /**
+   * Callback to change the active state of the Modal.
+   *
+   */
+  setActive?: (
+    active: boolean,
+  ) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback to determine whether or not Modal should close when user tries to close it.
+   *
+   */
+  modalShouldClose?: () => boolean;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  active
+  size="large"
+  setActive={setActive}
+  modalShouldClose={() => console.log('Modal is closing now!')}
+  >  
+  Modal content!
+</Modal>
+```
+ * @param props.active Boolean to describe whether or not Modal is active.
+ * @param props.size String to determine size of Modal. ['xxsmall', 'xsmall', 'small', 'default', 'large', 'xlarge']
+ * @param props.setActive Callback to change the active state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.modalShouldClose Callback to determine whether or not Modal should close when user tries to close it.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  active = false,
+  size = ModalSize.Default,
+  setActive,
+  children,
+  modalShouldClose,
+  className,
+  ...rest
+}: ModalProps) {
+  if (!active) {
+    return null;
+  }
+
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = () => {
+    // Don't close modal if modalShouldClose returns false or no setActive callback is passed
+    if (!setActive || (modalShouldClose && !modalShouldClose())) {
+      return;
+    }
+
+    setActive(false);
+  };
+
+  const handleDocumentClick = (e: React.SyntheticEvent) => {
+    if (!contentRef.current) {
+      return;
+    }
+
+    if (!contentRef.current.contains(e.target as Node)) {
+      handleClose();
+    }
+  };
+
+  const handleEscape = useCallback((e: KeyboardEvent) => {
+    if (e.keyCode === escapeKey) {
+      handleClose();
+    }
+  }, []);
+
+  useEventListener('keydown', handleEscape, {
+    options: { once: true },
+  });
+
+  return (
+    <Portal>
+      <Global
+        styles={`
+        html, body {
+          overflow: hidden;
+        }
+
+        body {
+          position: relative;
+        }
+      `}
+      />
+
+      <div {...rest} onClick={handleDocumentClick} className={scrollContainer}>
+        <div
+          className={cx(modalContentStyle, modalSizes[size], className)}
+          tabIndex={-1}
+          ref={contentRef}
+        >
+          <Icon
+            glyph="X"
+            fill={'#5D6C74'}
+            size={Size.Large}
+            onClick={handleClose}
+            className={closeButton}
+            data-dismiss="modal"
+            aria-hidden="true"
+          />
+
+          <div className={modalBodyStyle}>{children}</div>
```

Comment:
I feel there will be a day when someone wants to set the className on this div.

Reply:
should we expose a modalBodyClassName prop? @DesignerDave

Reply:
updated such that we no longer need this prop. className is being spread properly on the container.

Reply:
As a general rule, we should avoid adding props like that, especially before they're asked for.